src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -91,10 +91,35 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
         // tools for loading the compiled syntax tree from the dll resource (later setup for shipping Q# libraries)
 
         /// <summary>
-        /// Given a stream containing the binary representation of compiled Q# code, returns the corresponding Q# compilation.
-        /// Returns true if the compilation could be deserialized without throwing an exception, and false otherwise.
+        /// Given a binary representation of compiled Q# code, returns the corresponding Q# compilation.
+        /// Returns true if the compilation could be deserialized without throwing an exception, and it is not empty. False otherwise.
         /// If onDeserializationException is specified, invokes the given action on any exception thrown during deserialization.
         /// </summary>
+        public static bool LoadSyntaxTree(
+            byte[] byteArray,
+            [NotNullWhen(true)] out QsCompilation? compilation,
+            Action<Exception>? onDeserializationException = null)
+        {
+            compilation = null;
+            try
+            {
+                compilation = BondSchemas.Protocols.DeserializeQsCompilationFromSimpleBinary(byteArray);
+            }
+            catch (Exception ex)
+            {
+                onDeserializationException?.Invoke(ex);
+                return false;
+            }
+
+            return compilation != null && IsValidCompilation(compilation);
+        }
+
+        /// <summary>
+        /// Given a stream containing a binary representation of compiled Q# code, returns the corresponding Q# compilation.
+        /// Returns true if the compilation could be deserialized without throwing an exception, and it is not empty. False otherwise.
```

Comment:
```suggestion
        /// Returns true if the compilation could be deserialized without throwing an exception, and it is properly instantiated. False otherwise.
```